.pre-commit-config.yaml

```diff
@@ -0,0 +1,23 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+---
+repos:
+  - repo: local
+    hooks:
+      - id: run-codebase-tests
+        name: Run Pytest on codebase tests
+        description: This hook checks for codebase tests.
+        entry: pytest tests/codebase
+        language: conda
+        # Mandatory otherwise `pre-commit` passes the names of all files to be commited
+        pass_filenames: false
+        stages: [push]
+
+      - id: protect-branches
+        name: Protect Git branches
+        description: This hook avoid pushing to the protected Git branches.
+        entry: python scripts/protect_branch_git_hook.py
```

Comment:
why don't we make a `hooks` subdirectory then `python scripts/hooks/protect_branches.py` (I expect we will want more in the future, might as well add a place to group them now)